test/Vault.spec.ts

```diff
@@ -501,61 +501,116 @@ describe("Vault", () => {
 
     it("reverts if no investable amount", async () => {
       await vault.connect(owner).setStrategy(strategy.address);
-      await addYieldToVault("10");
```

Comment:
Why did you take it out? Shouldn't this test still work without changes?

Reply:
in the old version of this PR  (`disinvest` branch) this still worked because in the test the `invest` function was called directly

when we use updateInvested and run addYieldToVault('10') beforehand, the updateInvested function disinvests instead of investing so the assertion at the bottom of the test fails